exercises/atbash-cipher/example.rs

```diff
@@ -0,0 +1,39 @@
+fn ascii(ch: char) -> u8 {
+    ch as u8
+}
+
+fn get_transpose(ch: char) -> char {
+    if ch.is_digit(10) {
+        ch
+    } else {
+        (ascii('z') - ascii(ch) + ascii('a')) as char
+    }
+}
+
+pub fn encode(plaintext: &str) -> String {
+    plaintext
+    .to_lowercase()
+    .chars()
+    .filter(|&ch| ch.is_alphanumeric())
+    .filter(|&ch| (ch as u8) < 128)
+    .map(|ch| get_transpose(ch))
+    .fold("".to_string(),|encoded_result, ch| {
+        if (encoded_result.len() + 1) % 6 == 0 {
+            encoded_result + " " + &(ch.to_string())
+        } else {
+            encoded_result + &(ch.to_string())
+        }
+    })
+}
+
+pub fn decode(ciphertext: &str) -> String {
+    ciphertext
+    .chars()
+    .fold("".to_string(), |decoded_result, ch| {
```

Comment:
I think doing a filter to remove spaces & then a map to transpose would be more idiomatic. Saves you the `if`, at least.